### ISSUE TITLE
Wiki Sidebar None not found

### ISSUE THREAD

**Issue by saru2020** [frappe/wiki#134]
master branch is throwing below error:

> Executing wiki.wiki.doctype.wiki_settings.patches.wiki_sidebar_migration in mysite.localhost (_17bae7aaa9b40f3d)
Queued rebuilding of search index for mysite.localhost
Traceback with variables (most recent call last):
  File "/home/frappe/.pyenv/versions/3.10.5/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
      mod_name = 'frappe.utils.bench_helper'
      alter_argv = True
      mod_spec = ModuleSpec(name='frappe.utils.bench_helper', loader=<_frozen_importlib_external.SourceFileLoader object at 0xffffb8bcb1c0>, origin='/workspace/development/frappe-bench/apps/frappe/frappe/utils/bench_helper.py')
      code = <code object <module> at 0xffffb69354d0, file "/workspace/development/frappe-bench/apps/frappe/frappe/utils/bench_helper.py", line 1>
      main_globals = {'__name__': '__main__', '__doc__': None, '__package__': 'frappe.utils', '__loader__': <_frozen_importlib_external.SourceFileLoader object at 0xffffb8bcb1c0>, '__spec__': ModuleSpec(name='frappe.utils.bench_helper', loader=<_frozen_importlib_external.SourceFileLoader object at 0xffffb8bcb1c0>, origin='/workspace/development/frappe-bench/apps/frappe/frappe/utils/bench_helper.py'), '__annotations__': {}, '__builtins__': <module 'builtins' (built-in)>, '__file__': '/workspace/development/frappe-bench/apps/frappe/frappe/utils/bench_helper.py', '__cached__': '/workspace/development/frappe-bench/apps/frappe/frappe/utils/__pycache__/bench_helper.cpython-310.pyc', 'importlib': <module 'importlib' from '/home/frappe/.pyenv/versions/3.10.5/lib/python3.10/importlib/__init__.py'>, 'json': <module 'json' from '/home/frappe/.pyenv/versions/3.10.5/lib/python3.10/json/__init__.py'>, 'os': <module 'os' from '/home/frappe/.pyenv/versions/3.10.5/lib/python3.10/os.py'>, 'traceback': <module 'traceback' fr...
  File "/home/frappe/.pyenv/versions/3.10.5/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
      code = <code object <module> at 0xffffb69354d0, file "/workspace/development/frappe-bench/apps/frappe/frappe/utils/bench_helper.py", line 1>
      run_globals = {'__name__': '__main__', '__doc__': None, '__package__': 'frappe.utils', '__loader__': <_frozen_importlib_external.SourceFileLoader object at 0xffffb8bcb1c0>, '__spec__': ModuleSpec(name='frappe.utils.bench_helper', loader=<_frozen_importlib_external.SourceFileLoader object at 0xffffb8bcb1c0>, origin='/workspace/development/frappe-bench/apps/frappe/frappe/utils/bench_helper.py'), '__annotations__': {}, '__builtins__': <module 'builtins' (built-in)>, '__file__': '/workspace/development/frappe-bench/apps/frappe/frappe/utils/bench_helper.py', '__cached__': '/workspace/development/frappe-bench/apps/frappe/frappe/utils/__pycache__/bench_helper.cpython-310.pyc', 'importlib': <module 'importlib' from '/home/frappe/.pyenv/versions/3.10.5/lib/python3.10/importlib/__init__.py'>, 'json': <module 'json' from '/home/frappe/.pyenv/versions/3.10.5/lib/python3.10/json/__init__.py'>, 'os': <module 'os' from '/home/frappe/.pyenv/versions/3.10.5/lib/python3.10/os.py'>, 'traceback': <module 'traceback' fr...
      init_globals = None
      mod_name = '__main__'
      mod_spec = ModuleSpec(name='frappe.utils.bench_helper', loader=<_frozen_importlib_external.SourceFileLoader object at 0xffffb8bcb1c0>, origin='/workspace/development/frappe-bench/apps/frappe/frappe/utils/bench_helper.py')
      pkg_name = 'frappe.utils'
      script_name = None
      loader = <_frozen_importlib_external.SourceFileLoader object at 0xffffb8bcb1c0>
      fname = '/workspace/development/frappe-bench/apps/frappe/frappe/utils/bench_helper.py'
      cached = '/workspace/development/frappe-bench/apps/frappe/frappe/utils/__pycache__/bench_helper.cpython-310.pyc'
  File "/workspace/development/frappe-bench/apps/frappe/frappe/utils/bench_helper.py", line 104, in <module>
    main()
      ...skipped... 26 vars
  File "/workspace/development/frappe-bench/apps/frappe/frappe/utils/bench_helper.py", line 19, in main
    click.Group(commands=commands)(prog_name="bench")
      commands = {'frappe': <Group frappe>, 'get-frappe-commands': <Command get-frappe-commands>, 'get-frappe-help': <Command get-frappe-help>}
  File "/workspace/development/frappe-bench/env/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
      self = <Group None>
      args = ()
      kwargs = {'prog_name': 'bench'}
  File "/workspace/development/frappe-bench/env/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
      self = <Group None>
      args = ['frappe', '--site', 'mysite.localhost', 'migrate']
      prog_name = 'bench'
      complete_var = None
      standalone_mode = True
      windows_expand_args = True
      extra = {}
      ctx = <click.core.Context object at 0xffffb691ee90>
  File "/workspace/development/frappe-bench/env/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
      _process_result = <function MultiCommand.invoke.<locals>._process_result at 0xffffb6968940>
      args = ['migrate']
      cmd_name = 'frappe'
      cmd = <Group frappe>
      sub_ctx = <click.core.Context object at 0xffffb698b070>
      ctx = <click.core.Context object at 0xffffb691ee90>
      self = <Group None>
      __class__ = <class 'click.core.MultiCommand'>
  File "/workspace/development/frappe-bench/env/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
      _process_result = <function MultiCommand.invoke.<locals>._process_result at 0xffffb69408b0>
      args = []
      cmd_name = 'migrate'
      cmd = <Command migrate>
      sub_ctx = <click.core.Context object at 0xffffb698a050>
      ctx = <click.core.Context object at 0xffffb698b070>
      self = <Group frappe>
      __class__ = <class 'click.core.MultiCommand'>
  File "/workspace/development/frappe-bench/env/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
      self = <Command migrate>
      ctx = <click.core.Context object at 0xffffb698a050>
  File "/workspace/development/frappe-bench/env/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
      _Context__self = <click.core.Context object at 0xffffb698a050>
      _Context__callback = <function migrate at 0xffffb69ac700>
      args = ()
      kwargs = {'skip_failing': False, 'skip_search_index': False}
      ctx = <click.core.Context object at 0xffffb698a050>
  File "/workspace/development/frappe-bench/env/lib/python3.10/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
      args = ()
      kwargs = {'skip_failing': False, 'skip_search_index': False}
      f = <function migrate at 0xffffb69ac4c0>
  File "/workspace/development/frappe-bench/apps/frappe/frappe/commands/__init__.py", line 29, in _func
    ret = f(frappe._dict(ctx.obj), *args, **kwargs)
      ctx = <click.core.Context object at 0xffffb698a050>
      args = ()
      kwargs = {'skip_failing': False, 'skip_search_index': False}
      profile = False
      f = <function migrate at 0xffffb69ac430>
  File "/workspace/development/frappe-bench/apps/frappe/frappe/commands/site.py", line 605, in migrate
    SiteMigration(
      context = {'sites': ['mysite.localhost'], 'force': False, 'verbose': False, 'profile': False}
      skip_failing = False
      skip_search_index = False
      activate_by_import = <module 'traceback_with_variables.activate_by_import' from '/workspace/development/frappe-bench/env/lib/python3.10/site-packages/traceback_with_variables/activate_by_import.py'>
      SiteMigration = <class 'frappe.migrate.SiteMigration'>
      site = 'mysite.localhost'
  File "/workspace/development/frappe-bench/apps/frappe/frappe/migrate.py", line 178, in run
    self.run_schema_updates()
      self = <frappe.migrate.SiteMigration object at 0xffffb6989f30>
      site = 'mysite.localhost'
      filelock = <function filelock at 0xffffb6969fc0>
  File "/workspace/development/frappe-bench/apps/frappe/frappe/migrate.py", line 41, in wrapper
    ret = method(*args, **kwargs)
      args = (<frappe.migrate.SiteMigration object at 0xffffb6989f30>,)
      kwargs = {}
      method = <function SiteMigration.run_schema_updates at 0xffffb47c6050>
  File "/workspace/development/frappe-bench/apps/frappe/frappe/migrate.py", line 114, in run_schema_updates
    frappe.modules.patch_handler.run_all(
      self = <frappe.migrate.SiteMigration object at 0xffffb6989f30>
  File "/workspace/development/frappe-bench/apps/frappe/frappe/modules/patch_handler.py", line 75, in run_all
    run_patch(patch)
      patch_type = <PatchType.post_model_sync: 'post_model_sync'>
      executed = {"execute:frappe.reload_doc('core', 'doctype', 'version') #2017-04-01", 'frappe.patches.v14_0.setup_likes_from_feedback', 'wiki.wiki.doctype.wiki_sidebar_item.patches.fetch_route', 'frappe.patches.v13_0.remove_share_for_std_users', 'frappe.patches.v14_0.remove_post_and_post_comment', 'frappe.patches.v12_0.reset_home_settings', "execute:frappe.delete_doc_if_exists('DocType', 'GSuite Templates')", 'frappe.patches.v15_0.remove_prepared_report_settings_from_system_settings', "execute:frappe.reload_doc('core', 'doctype', 'error_snapshot')", 'frappe.patches.v13_0.email_unsubscribe', 'frappe.patches.v11_0.remove_skip_for_doctype', 'frappe.patches.v11_0.rename_google_maps_doctype', 'frappe.patches.v13_0.set_path_for_homepage_in_web_page_view', 'frappe.patches.v10_0.enhance_security', 'frappe.patches.v13_0.remove_chat', "execute:frappe.delete_doc_if_exists('DocType', 'GCalendar Account')", 'frappe.patches.v14_0.drop_data_import_legacy', 'execute:frappe.delete_doc("Page", "activity", force=1)', ...
      run_patch = <function run_all.<locals>.run_patch at 0xffffb442f400>
      patches = ['frappe.core.doctype.role.patches.v13_set_default_desk_properties', 'frappe.patches.v14_0.drop_data_import_legacy', 'frappe.patches.v14_0.copy_mail_data #08.03.21', 'frappe.patches.v14_0.update_github_endpoints #08-11-2021', 'frappe.patches.v14_0.remove_db_aggregation', 'frappe.patches.v14_0.update_color_names_in_kanban_board_column', 'frappe.patches.v14_0.update_is_system_generated_flag', 'frappe.patches.v14_0.update_auto_account_deletion_duration', 'frappe.patches.v14_0.update_integration_request', 'frappe.patches.v14_0.set_document_expiry_default', 'frappe.patches.v14_0.delete_data_migration_tool', 'frappe.patches.v14_0.set_suspend_email_queue_default', 'frappe.patches.v14_0.different_encryption_key', 'frappe.patches.v14_0.update_multistep_webforms', "execute:frappe.delete_doc('Page', 'background_jobs', ignore_missing=True, force=True)", 'frappe.patches.v14_0.drop_unused_indexes', 'frappe.patches.v15_0.drop_modified_index', 'frappe.patches.v14_0.update_attachment_comment', 'frappe....
      patch = 'wiki.wiki.doctype.wiki_settings.patches.wiki_sidebar_migration'
      skip_failing = False
  File "/workspace/development/frappe-bench/apps/frappe/frappe/modules/patch_handler.py", line 62, in run_patch
    if not run_single(patchmodule=patch):
      patch = 'wiki.wiki.doctype.wiki_settings.patches.wiki_sidebar_migration'
      skip_failing = False
  File "/workspace/development/frappe-bench/apps/frappe/frappe/modules/patch_handler.py", line 150, in run_single
    return execute_patch(patchmodule, method, methodargs)
      patchmodule = 'wiki.wiki.doctype.wiki_settings.patches.wiki_sidebar_migration'
      method = None
      methodargs = None
      force = False
      conf = <LocalProxy unbound>
  File "/workspace/development/frappe-bench/apps/frappe/frappe/modules/patch_handler.py", line 186, in execute_patch
    _patch()
      patchmodule = 'wiki.wiki.doctype.wiki_settings.patches.wiki_sidebar_migration'
      method = None
      methodargs = None
      has_patch_file = True
      patch = 'wiki.wiki.doctype.wiki_settings.patches.wiki_sidebar_migration.execute'
      docstring = ''
      _patch = <function execute at 0xffffb4314310>
      start_time = 10514.487714122
  File "/workspace/development/frappe-bench/apps/wiki/wiki/wiki/doctype/wiki_settings/patches/wiki_sidebar_migration.py", line 19, in execute
    sidebar_items = get_children(frappe.get_doc("Wiki Sidebar", topmost))
      wiki_settings = <WikiSettings: Wiki Settings>
      topmost = None
  File "/workspace/development/frappe-bench/apps/frappe/frappe/__init__.py", line 1168, in get_doc
    doc = frappe.model.document.get_doc(*args, **kwargs)
      args = ('Wiki Sidebar', None)
      kwargs = {}
      frappe = <module 'frappe' from '/workspace/development/frappe-bench/apps/frappe/frappe/__init__.py'>
  File "/workspace/development/frappe-bench/apps/frappe/frappe/model/document.py", line 75, in get_doc
    return controller(*args, **kwargs)
      args = ('Wiki Sidebar', None)
      kwargs = {}
      doctype = 'Wiki Sidebar'
      controller = <class 'wiki.wiki.doctype.wiki_sidebar.wiki_sidebar.WikiSidebar'>
  File "/workspace/development/frappe-bench/apps/frappe/frappe/model/document.py", line 107, in __init__
    self.load_from_db()
      self = <WikiSidebar: unsaved>
      args = ('Wiki Sidebar', None)
      kwargs = {}
      __class__ = <class 'frappe.model.document.Document'>
  File "/workspace/development/frappe-bench/apps/frappe/frappe/model/document.py", line 159, in load_from_db
    frappe.throw(
      self = <WikiSidebar: unsaved>
      get_value_kwargs = {'for_update': None, 'as_dict': True, 'order_by': None}
      d = None
      __class__ = <class 'frappe.model.document.Document'>
  File "/workspace/development/frappe-bench/apps/frappe/frappe/__init__.py", line 523, in throw
    msgprint(
      msg = 'Wiki Sidebar None not found'
      exc = <class 'frappe.exceptions.DoesNotExistError'>
      title = None
      is_minimizable = False
      wide = False
      as_list = False
  File "/workspace/development/frappe-bench/apps/frappe/frappe/__init__.py", line 491, in msgprint
    _raise_exception()
      title = None
      as_table = False
      as_list = False
      indicator = 'red'
      alert = False
      primary_action = None
      is_minimizable = False
      wide = False
      sys = <module 'sys' (built-in)>
      out = {'message': 'Wiki Sidebar None not found', 'title': 'Message', 'indicator': 'red', 'raise_exception': 1}
      _raise_exception = <function msgprint.<locals>._raise_exception at 0xffffb45ca0e0>
      _strip_html_tags = <functools._lru_cache_wrapper object at 0xffffb45d4300>
      msg = 'Wiki Sidebar None not found'
      raise_exception = <class 'frappe.exceptions.DoesNotExistError'>
      strip_html_tags = <function strip_html_tags at 0xffffb79ca4d0>
      ...skipped... 1 vars
  File "/workspace/development/frappe-bench/apps/frappe/frappe/__init__.py", line 440, in _raise_exception
    raise raise_exception(msg)
      msg = 'Wiki Sidebar None not found'
      raise_exception = <class 'frappe.exceptions.DoesNotExistError'>
      ...skipped... 1 vars
frappe.exceptions.DoesNotExistError: Wiki Sidebar None not found

`bench --site mysite.localhost list-apps`

frappe     15.x.x-develop () develop
wiki       0.0.1             master
payments   0.0.1             develop

**Comment by BreadGenie** [comment]
Can you set the sidebar field in Wiki Settings to the root sidebar from Wiki Sidebar (if you can't find the root sidebar just add a sidebar that exists in Wiki Sidebar), since it's necessary for migration?

**Comment by saru2020** [comment]
I'm not able to run the app at all.

I tried uninstalling the app and re-installing and it works but my restored data was gone but the app was working. now, If I try to restore my previous site, its throwing this error:

> AttributeError: 'WebsiteSettings' object has no attribute 'splash_image'

**Comment by BreadGenie** [comment]
Can you post the traceback?

**Comment by saru2020** [comment]
> 10:57:58 web.1      | /workspace/development/frappe-bench/apps/frappe/frappe/utils/caching.py:57: DeprecationWarning: `frappe_last` argument is deprecated and will be removed in v15.
10:57:58 web.1      |   return_val = func(*args, **kwargs)
10:58:01 web.1      | 172.18.0.1 - - [08/Apr/2023 10:58:01] "GET / HTTP/1.1" 500 -
10:58:01 web.1      | Traceback (most recent call last):
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/serve.py", line 18, in get_response
10:58:01 web.1      |     response = renderer_instance.render()
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/page_renderers/template_page.py", line 84, in render
10:58:01 web.1      |     html = self.get_html()
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/utils.py", line 510, in cache_html_decorator
10:58:01 web.1      |     html = func(*args, **kwargs)
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/page_renderers/template_page.py", line 92, in get_html
10:58:01 web.1      |     self.init_context()
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/page_renderers/base_template_page.py", line 15, in init_context
10:58:01 web.1      |     self.context.update(get_website_settings())
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/doctype/website_settings/website_settings.py", line 203, in get_website_settings
10:58:01 web.1      |     if settings.splash_image:
10:58:01 web.1      | AttributeError: 'WebsiteSettings' object has no attribute 'splash_image'
10:58:01 web.1      | 
10:58:01 web.1      | During handling of the above exception, another exception occurred:
10:58:01 web.1      | 
10:58:01 web.1      | Traceback (most recent call last):
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/app.py", line 75, in application
10:58:01 web.1      |     response = get_response()
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/serve.py", line 27, in get_response
10:58:01 web.1      |     response = ErrorPage(exception=e).render()
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/page_renderers/template_page.py", line 84, in render
10:58:01 web.1      |     html = self.get_html()
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/utils.py", line 510, in cache_html_decorator
10:58:01 web.1      |     html = func(*args, **kwargs)
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/page_renderers/template_page.py", line 92, in get_html
10:58:01 web.1      |     self.init_context()
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/page_renderers/error_page.py", line 14, in init_context
10:58:01 web.1      |     super().init_context()
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/page_renderers/base_template_page.py", line 15, in init_context
10:58:01 web.1      |     self.context.update(get_website_settings())
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/doctype/website_settings/website_settings.py", line 203, in get_website_settings
10:58:01 web.1      |     if settings.splash_image:
10:58:01 web.1      | AttributeError: 'WebsiteSettings' object has no attribute 'splash_image'
10:58:01 web.1      | 
10:58:01 web.1      | During handling of the above exception, another exception occurred:
10:58:01 web.1      | 
10:58:01 web.1      | Traceback (most recent call last):
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/serve.py", line 18, in get_response
10:58:01 web.1      |     response = renderer_instance.render()
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/page_renderers/template_page.py", line 84, in render
10:58:01 web.1      |     html = self.get_html()
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/utils.py", line 510, in cache_html_decorator
10:58:01 web.1      |     html = func(*args, **kwargs)
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/page_renderers/template_page.py", line 92, in get_html
10:58:01 web.1      |     self.init_context()
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/page_renderers/base_template_page.py", line 15, in init_context
10:58:01 web.1      |     self.context.update(get_website_settings())
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/doctype/website_settings/website_settings.py", line 203, in get_website_settings
10:58:01 web.1      |     if settings.splash_image:
10:58:01 web.1      | AttributeError: 'WebsiteSettings' object has no attribute 'splash_image'
10:58:01 web.1      | 
10:58:01 web.1      | During handling of the above exception, another exception occurred:
10:58:01 web.1      | 
10:58:01 web.1      | Traceback (most recent call last):
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/middlewares.py", line 16, in __call__
10:58:01 web.1      |     return super().__call__(environ, start_response)
10:58:01 web.1      |   File "/workspace/development/frappe-bench/env/lib/python3.10/site-packages/werkzeug/middleware/shared_data.py", line 247, in __call__
10:58:01 web.1      |     return self.app(environ, start_response)
10:58:01 web.1      |   File "/workspace/development/frappe-bench/env/lib/python3.10/site-packages/werkzeug/middleware/shared_data.py", line 247, in __call__
10:58:01 web.1      |     return self.app(environ, start_response)
10:58:01 web.1      |   File "/workspace/development/frappe-bench/env/lib/python3.10/site-packages/werkzeug/local.py", line 237, in application
10:58:01 web.1      |     return ClosingIterator(app(environ, start_response), self.cleanup)
10:58:01 web.1      |   File "/workspace/development/frappe-bench/env/lib/python3.10/site-packages/werkzeug/wrappers/request.py", line 194, in application
10:58:01 web.1      |     resp = f(*args[:-2] + (request,))
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/app.py", line 84, in application
10:58:01 web.1      |     response = handle_exception(e)
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/app.py", line 324, in handle_exception
10:58:01 web.1      |     response = get_response("message", http_status_code=http_status_code)
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/serve.py", line 27, in get_response
10:58:01 web.1      |     response = ErrorPage(exception=e).render()
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/page_renderers/template_page.py", line 84, in render
10:58:01 web.1      |     html = self.get_html()
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/utils.py", line 510, in cache_html_decorator
10:58:01 web.1      |     html = func(*args, **kwargs)
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/page_renderers/template_page.py", line 92, in get_html
10:58:01 web.1      |     self.init_context()
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/page_renderers/error_page.py", line 14, in init_context
10:58:01 web.1      |     super().init_context()
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/page_renderers/base_template_page.py", line 15, in init_context
10:58:01 web.1      |     self.context.update(get_website_settings())
10:58:01 web.1      |   File "/workspace/development/frappe-bench/apps/frappe/frappe/website/doctype/website_settings/website_settings.py", line 203, in get_website_settings
10:58:01 web.1      |     if settings.splash_image:
10:58:01 web.1      | AttributeError: 'WebsiteSettings' object has no attribute 'splash_image'

**Comment by BreadGenie** [comment]
Can you try restoring first and then updating Wiki? The traceback seems to be originating from website module and not Wiki.

**Comment by saru2020** [comment]
I tried switching the frappe branch from `develop` to `version-14` and then did a `bench restore` and then ran `bench migrate` and I saw this issue: 
> AttributeError: 'WebsiteSettings' object has no attribute 'splash_image'

but when I try to restore the site and then do `bench update --reset --no-backup`, then it takes a very long time at the wiki app installation step and then it throws this error:
> frappe.exceptions.DoesNotExistError: Wiki Sidebar None not found


how do i resolve this?

**Comment by saru2020** [comment]
> Can you set the sidebar field in Wiki Settings to the root sidebar from Wiki Sidebar (if you can't find the root sidebar just add a sidebar that exists in Wiki Sidebar), since it's necessary for migration?

I tried installing everything from scratch on a fresh Ubuntu machine and was able to run the app (except wiki).
from Wiki Settings, I tried to find the root sidebar but it didn't exist so added a custom/existing sidebar, and the wiki app is working now.
would there be a permanent fix for this issue?

**Comment by BreadGenie** [comment]
> > Can you set the sidebar field in Wiki Settings to the root sidebar from Wiki Sidebar (if you can't find the root sidebar just add a sidebar that exists in Wiki Sidebar), since it's necessary for migration?
> 
> I tried installing everything from scratch on a fresh Ubuntu machine and was able to run the app (except wiki). from Wiki Settings, I tried to find the root sidebar but it didn't exist so added a custom/existing sidebar, and the wiki app is working now. would there be a permanent fix for this issue?

What I meant as root sidebar is a `Wiki Sidebar` record with a table having the root level sidebars you see in the wiki page and it's not a name (by default the name is just "wiki").
example
```
├── Group 1
│   ├──Group 2
│   │   ├── Wiki Page 1
│   ├── Wiki Page 2
├── Group 3
│   ├──Wiki Page 2
```

here the sidebar with Group 1 and Group 3 in the table will be the root sidebar. (After the rewrite this isn't applicable as sidebars are stored in `Wiki Settings` from now on.)

> edit: also, now the path of all the wiki pages needs to be updated, would there be any fix to address this as well?

Sorry I didn't get what you meant here? Did you want to bulk rename the route? If it's because it's prepended by `/wiki`, you can go to `Wiki Settings` and set the scope as empty.

**Comment by saru2020** [comment]
> What I meant as root sidebar is a `Wiki Sidebar` record with a table having the root level sidebars you see in the wiki page and it's not a name (by default the name is just "wiki"). for example, in your wiki page if the sidebar looks like:
> 
> ```
> ├── Group 1
> │   ├──Group 2
> │   │   ├── Wiki Page 1
> │   ├── Wiki Page 2
> ├── Group 3
> │   ├──Wiki Page 3
> ```
> 
> here the sidebar with Group 1 and Group 3 in the table will be the root sidebar. (After the rewrite this isn't applicable as sidebars are stored in `Wiki Settings` from now on.)

I think this is the problem. in the earlier version, there was no root sidebar at all, when I hit this url: "\<domain\>/wiki", it'd redirect to the first wiki page itself. from your example, to 'Wiki Page 1', for example, if 'Wiki Page 1' had a route as 'wiki_page_1', then when we try to visit '<domain>/wiki', it'd redirect to '\<domain\>/wiki_page_1' (this is the behavior of the previous version of the app)

> Sorry I didn't get what you meant here? Did you want to bulk rename the route? If it's because it's prepended by /wiki, you can go to Wiki Settings and set the wiki_search_scope as empty.
Edit: leaving it empty might give you irrelevant search results.

this is relevant to the route/path/url of the Wiki pages.
when I manually created a side bar item named 'root' as mentioned [here](https://github.com/frappe/wiki/issues/134#issuecomment-1500945332) (Question: would there be any fix for this issue?) and then mapped the other Wiki sidebar to it, visiting any of those Wiki pages has url like this: '\<domain\>/**None**/wiki_page_1 (updated Wiki app, 'None' was prepended to the Wiki page route) instead of '\<domain\>/wiki_page_1' (behavior of the previous version of the app)

note: you might have to install the previous version and restore the old site in order to replicate this issue.

**Comment by BreadGenie** [comment]
I think it's better off for you to use legacy branch until the newer wiki is stable (there'll be even more breaking changes).

**Comment by BreadGenie** [comment]
Can you try again with the updated version-14 branch now?
The issue is resolved in #139 (specifically b4eceaabc47ffad52f48536c98040ed266d6eda6)

**Comment by BreadGenie** [comment]
Closing since the issue is fixed but feel free to reopen if it persist for you
